source/auth.c

```diff
@@ -187,6 +203,250 @@ static napi_value s_creds_provider_new_static(napi_env env, const struct aws_nap
+    }
+
+    if (AWS_NGNPR_VALID_VALUE != aws_napi_get_named_property_as_bytebuf(
+                                     env, node_config, AWS_NAPI_KEY_IDENTITY, napi_string, &config->identity)) {
+        AWS_LOGF_ERROR(
+            AWS_LS_NODEJS_CRT_GENERAL,
+            "s_aws_cognito_credentials_provider_config_init - required property 'identity' could not be extracted from "
+            "config");
+        return aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
+    }
+
+    napi_value napi_logins = NULL;
+    if (AWS_NGNPR_VALID_VALUE ==
+        aws_napi_get_named_property(env, node_config, AWS_NAPI_KEY_LOGINS, napi_object, &napi_logins)) {
+
+        /* how many login entries */
+        uint32_t login_count = 0;
+        AWS_NAPI_CALL(env, napi_get_array_length(env, napi_logins, &login_count), {
+            AWS_LOGF_ERROR(
+                AWS_LS_NODEJS_CRT_GENERAL,
+                "s_aws_cognito_credentials_provider_config_init - property 'logins' must be an array");
+            return aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
+        });
+
+        for (size_t i = 0; i < login_count; ++i) {
+
+            napi_value napi_token_pair = NULL;
+            AWS_NAPI_CALL(env, napi_get_element(env, napi_logins, i, &napi_token_pair), {
+                AWS_LOGF_ERROR(
+                    AWS_LS_NODEJS_CRT_GENERAL,
+                    "s_aws_cognito_credentials_provider_config_init - could not access property 'logins' array "
+                    "element");
+                aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
+                goto done;
+            });
+
+            if (AWS_NGNPR_VALID_VALUE != aws_napi_get_named_property_as_bytebuf(
+                                             env,
+                                             napi_token_pair,
+                                             AWS_NAPI_KEY_IDENTITY_PROVIDER_NAME,
+                                             napi_string,
+                                             &identity_provider_name_buf)) {
+                AWS_LOGF_ERROR(
+                    AWS_LS_NODEJS_CRT_GENERAL,
+                    "s_aws_cognito_credentials_provider_config_init - required property 'identityProviderName' missing "
+                    "from login token pair");
+                aws_raise_error(AWS_ERROR_INVALID_ARGUMENT);
+                goto done;
+            }
+
+            if (AWS_NGNPR_VALID_VALUE != aws_napi_get_named_property_as_bytebuf(
+                                             env,
+                                             napi_token_pair,
+                                             AWS_NAPI_KEY_IDENTITY_PROVIDER_TOKEN,
+                                             napi_string,
+                                             &identity_provider_token_buf)) {
+                AWS_LOGF_ERROR(
+                    AWS_LS_NODEJS_CRT_GENERAL,
+                    "s_aws_cognito_credentials_provider_config_init - required property 'identityProviderToken' "
```

Comment:
hmmm, I'd like to learn the reason about log out the function name here. My personal feeling is that the function name would only be helpful to maintainer (? We can always do search on the log info to locate the code). It's not helpful for customer?

Reply:
I may be misinterpreting, but I don't agree with the notion that "only information useful to a customer should go in logging"

Reply:
Yeah, but I also don't get how the name of the static function helps maintainer as well. 

Also, the details about which field or part from the Nodejs `CognitoCredentialsProviderConfig` are logged in C. The nodejs exception only has `Failed to initialize cognito provider configuration from node config`.

From the experience with Java SDK, I think we should try to have those debug info throw out to the language level. Like it's not a good experience to let nodejs user enable the C log and check the message out. 

So, I think maybe we can move those check of the configurations to nodejs level instead? 

On the other hand, I think yeah, maybe the `Failed to initialize cognito provider configuration from node config` is good enough for people to know to check their configuration and the document to get the idea. 🤷‍♂️

Reply:
Moving all the validation into the CRTs implies we're duplicating that validation across all CRTs rather than having it in one spot, which kind of goes against one of our guiding principles.

Reply:
Logging's primary purpose is to help us debug.  I find the function name helpful because it's an instant zero-in search term.  Otherwise you have to muddle around trying to figure out what is a sufficient search string.  Sometimes that's easy, sometimes it's a real pain, it all depends on the wording in the line (which is often fragmented by substitution parameters which disrupt the search).